Doc/Manual/Fortran.html

```diff
@@ -0,0 +1,2192 @@
+</p>
+
+
+<p>
+Because Fortran 2003 does specify support for a special <code>FINAL</code> procedure to clean up local or dynamic variables, the call to <code>release()</code> can be replaced by adding a <code>FINAL</code> procedure. The SWIG Fortran interface can generate this procedure, which will call the C++ destructor:
+</p>
+
+
+<div class="code"><pre><code>%feature(&quot;final&quot;) Foo;
+%include &quot;Foo.h&quot;
+</code></pre>
+</div>
+
+
+<p>
+<strong>However</strong>, this feature is relatively untested and its behavior could be compiler-dependent, so extreme caution is recommended when enabling it.
+</p>
+
+<H3><a name="Fortran_member_functions">Member functions</a></H3>
+
+<p>
+SWIG generates unique, private procedure names (generally <code>swigf_{classname}_{funcname}</code> for each class and function. These procedures are bound to the type. If <a href="#Fortran_function_overloading">function overloading</a> is used, “generic” procedures will be added to the derived type.
+</p>
+
+
+<p>
+Type-bound procedures in Fortran proxy classes are treated exactly the same as for native derived types:
+</p>
+
+
+<div class="targetlang"><pre><code>integer(C_INT) :: value
+type(Foo) :: food
+food = Foo()
+call food%do_something()
+value = food%get_something()
+</code></pre>
+</div>
+
+
+<p>
+Function overloading for derived types is implemented using <em>generic interfaces</em>. Each overloaded function gets a unique internal symname, and they are bound together in a generic interface. For example, if a member function <code>doit</code> of class <code>Action</code> is overloaded, a generic binding will be generated inside the Fortran proxy derived type:
+</p>
+
+
+<div class="targetlang"><pre><code>  procedure, private :: doit__SWIG_0 =&gt; swigf_Action_doit__SWIG_0
+  procedure, private :: doit__SWIG_1 =&gt; swigf_Action_doit__SWIG_1
+  generic :: doit =&gt; doit__SWIG_0, doit__SWIG_1
+</code></pre>
+</div>
+
+
+<p>
+As with <a href="#Fortran_function_overloading">free functions</a>, a member function returning <code>void</code> cannot be overloaded with a function returning non-void.
+</p>
+
+<H3><a name="Fortran_member_data">Member data</a></H3>
+
+<p>
+SWIG generates member functions for class member data in the same way that it generates free functions for <a href="#Fortran_global_variables">global variables</a>. Each public member produces a “getter”, and unless the data is marked <code>const</code>, it generates a “setter”.
+</p>
+
+
+<p>
+For a struct
+</p>
+
+<pre class="c++"><code>struct Foo {
+  int val;
+};
+</code></pre>
+
+<p>
+the interface to an instance and its data is:
+</p>
+
+
+<div class="targetlang"><pre><code>type(Foo) :: f
+f = Foo()
+call 
```

Comment:
The title is confusing as fragments are a term associated with typemaps. Please change the title to 'Code insertion blocks' and add in the % prefix like in SWIG.html#SWIG_nn42 and reference this section from somewhere below.

Reply:
I think the comment must be attached to the wrong code here -- I've changed the section titled "Fragments" to "Code insertion blocks".  I don't understand what you mean by "add in the % prefix" -- do you mean that I should provide macros `%fbegin`, `%fmodule`, etc. instead of requiring `%insert("fbegin")`, ...?

Reply:
Yes, you found the right place. The links aren't right though, I see:
```html
<a href="%22SWIG.html#SWIG_nn42%22">code insertion blocks</a>
```
Also, all the local href links are wrong in Fortran.html file. They need to changed for example, from:
```html
<a href="#Fortran_fundamental_types">Fundamental types</a>
```
to
```html
<a href="Fortran.htm#Fortran_fundamental_types">Fundamental types</a>
``` 
I was thinking you should use the same documentation style and conventions as is used in the main referred to section. For in the example, the heading in the table is "Name", when you mean "section". You have a table, whereas you could have copied the same style:
```
%fbegin %{
  ... code in fbegin section ...
%}
...etc...
``` 
These c++ sections have been expanded across all languages in the past, best to just refer to the main docs rather than repeat them in Fortran.html.